.github/workflows/my-ci-pipeline.yml

```diff
@@ -7,13 +7,44 @@ on:
     paths-ignore:
       - '**.md'
 jobs:
- build:
-  name: Build and test
-  runs-on: ubuntu-latest
-  steps:
-    - name: Checkout
-      uses: actions/checkout@v3
-    - name: Build
-      run: docker build --target development --tag todo_app:test .
-    - name: Test
-      run: docker run --env-file ./.env.test --mount type=bind,source="$(pwd)"/todo_app,target=/todo_app/todo_app --mount type=bind,source="$(pwd)"/tests,target=/todo_app/tests --entrypoint poetry todo_app:test run pytest
\ No newline at end of file
+  build:
+    name: Build and test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Build
+        run: docker build --target development --tag todo_app:test .
+      - name: Test
+        run: docker run --env-file ./.env.test --mount type=bind,source="$(pwd)"/todo_app,target=/todo_app/todo_app --mount type=bind,source="$(pwd)"/tests,target=/todo_app/tests --entrypoint poetry todo_app:test run pytest
+  deploy:
+    name: Deploy
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Login to Docker
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: Push image to Docker
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          target: production
+          push: true
+          tags: ${{ secrets.DOCKER_USERNAME }}/todo_app:latest, ${{ secrets.DOCKER_USERNAME }}/todo_app:${{ github.sha }}
+      - name: Login to Heroku
+        env:
+          HEROKU_API_KEY: ${{ secrets.HEROKU_API_KEY }}
```

Comment:
Personally I'd be tempted to pull this out to a job-level environment variable (e.g. unindent this and put it alongside - rather than in-the-middle-of - the `steps` field) so that we don't need to specify twice for separate steps, but just a preference